scripts/stage_layer.py

```diff
@@ -0,0 +1,94 @@
+
+
+class LayerBuilder:
+    layer_dir = Path(config.project_root) / 'lambdas' / 'layer'
+    out_dir = layer_dir / '.chalice' / 'terraform'
+
+    def __init__(self):
+        self.client = boto3.client('s3')
+
+    def update_layer_if_necessary(self, bucket, key):
+        log.info('Updating layer in bucket: %s with key %s', bucket, key)
+        if self.update_required(bucket, key):
+            log.info('Updating layer.zip')
+            self.update_layer(bucket, key)
+        else:
+            log.info('No update necessary')
+
+    def update_required(self, bucket, key) -> bool:
+        log.info('Checking if layer.zip needs to be updated')
+        try:
+            # Since the object key is content-addressable, just checking for
+            # the file's presence is sufficient
+            self.client.head_object(Bucket=bucket, Key=key)
+        except self.client.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                return True
+            else:
+                raise
+        else:
+            return False
+
+    def update_layer(self, bucket, key):
+        input_zip = self.out_dir / 'deployment.zip'
+        layer_zip = self.out_dir / 'layer.zip'
+        log.info('Tainting old lambda layer')
+        command = ['terraform', 'taint', 'aws_lambda_layer_version.dependencies']
+        subprocess.run(command, cwd=Path(config.project_root) / 'terraform')
+        self.build_layer_zip(layer_zip, input_zip)
+        log.info('Uploading layer zip to S3')
+        self.client.upload_file(str(layer_zip), bucket, key)
+
+    def build_layer_zip(self, destination, input_zip):
+        command = ['chalice', 'package', self.out_dir]
+        log.info('Running: %s', command)
+        subprocess.run(command, cwd=self.layer_dir).check_returncode()
+        log.info('Packaging %s', destination)
+        with ZipFile(input_zip, 'r') as deployment_zip:
+            with ZipFile(destination, 'w') as layer_zip:
+                for src_zip_info in deployment_zip.infolist():
+                    if src_zip_info.filename != 'app.py':
+                        # ZipFile doesn't copy permissions. Setting permissions
+                        # manually also requires setting other fields.
+                        dst_zip_info = ZipInfo(filename=str(Path('python') / src_zip_info.filename))
+                        dst_zip_info.external_attr = src_zip_info.external_attr
+                        dst_zip_info.date_time = src_zip_info.date_time
+                        dst_zip_info.compress_type = ZIP_DEFLATED
+                        with deployment_zip.open(src_zip_info, 'r') as rf:
+                            with layer_zip.open(dst_zip_info, 'w') as wf:
+                                wf.write(rf.read())
+
+
+def main(argv):
+    parser = ArgumentParser(description='Package the requirements layer')
+    parser.add_argument('--bucket', default=config.layer_bucket, help='bucket to hold layer file')
```

Comment:
While I see the point for `--force`, I don't think `--bucket` and `--key` are necessary.